### PR TITLE
just always load template tags to avoid problems in page builders

### PR DESCRIPTION
**Pull request by mnelson4**
## Problem this Pull Request solves

https://github.com/eventespresso/event-espresso-core/issues/360
Fixes a fatal error while using many page builder plugins resulting from template tags only being loaded on the front-end

## Why do you think these changes are needed in Event Espresso core instead of being put in an add-on?

Appropriate question for once. It was suggested this could be handled by an add-on or shim or code snippet, and we may go for that. But these are the reasons I thought it best to fix this issue in EE core and just always include the template tags, even on admin requests:
* loading the template tags file takes about 1 millisecond, and the conditional to determine if we should include it also takes about 1 second. Because it takes just as long to check as it does to just include it, no speed is saved by checking in the admin. In fact, it's a hair-width quicker to NOT check and just always include it
* it will be a bit of a pain to list out all the page builders out there, this avoids that
* when new page-builders are introduced, we don't have to include it in the whitelist

## How has this been tested

* [ ] Use elementor plugin, add an upcoming events widget to a page and try to save it. Upon reloading the page, I got the error from the issue. But on this branch it works fine
* [ ] Seth said he reproduced the issue in Beaver Builder (not sure what steps he took to reproduce)
* [ ] A general scan of the EE admin would be good too, just in case this somehow has unexpected consequences in there
* [ ] Test a theme that customizes EE's default templates. They should continue to work as normal.

## Checklist

* [x] I have added a changelog entry for this pull request
* [x] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [x] User input is adequately validated and sanitized
* [x] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [x] My code is tested.
* [x] My code follows the Event Espresso code style.
* [x] My code has proper inline documentation.
* [x] My code accounts for when the site is Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: core/EE_System.core.php [hunk 1]
```diff
@@ -1161,15 +1161,9 @@ public function core_loaded_and_ready()
...
-             || $this->request->isIframe()
-             || $this->request->isFeed()
-             ||  $this->request->isAjax()
-            ) && is_readable(EE_PUBLIC . 'template_tags.php')
-        ) {
-            require_once EE_PUBLIC . 'template_tags.php';
-        }
+        // always load template tags, because it's faster than checking if it's a front-end request, and many page
+        // builders require these even on the front-end
+        require_once EE_PUBLIC . 'template_tags.php';
```
Comment (nerrad) [inline core/EE_System.core.php]: should there still be a `is-readable` check though?

Comment (mnelson4) [reply core/EE_System.core.php]: good question. Under what circumstances, besides someone hacked EE core by removing or moving that file, would the file NOT be readable?
I see the constant `EE_PUBLIC` is defined in `core/espresso_definitions.php` and is always `define('EE_PUBLIC', EE_PLUGIN_DIR_PATH . 'public' . DS);`, where `EE_PLUGIN_DIR_PATH` is always the plugin's directory path. (Maybe at one point it was dynamic, but from the code I read it doesn't get much more static than this.)
From what I read, that file will always be present unless someone has hacked EE core. Right? If so, I'm inclined to not protect against someone hacking EE core, and save a bit of processing to check if the file exists.

Comment (nerrad) [reply core/EE_System.core.php]: works for me.

**Comment by mnelson4** [comment]
Passing to @joshfeck as you created the related issue (although you may prefer to pass onto Seth). I put some suggested testing notes under the title "How this has been tested" above

**Comment by joshfeck** [comment]
This fixes the issue reported earlier when trying to edit a page with Elementor, where the page has an embedded Event Espresso Upcoming Events widget.

**Pull request merged**